docs/src/userguide/language_basics.rst

```diff
@@ -119,40 +187,88 @@ You can read more about them in :ref:`extension-types`.
-Note that Cython uses array access for pointer dereferencing, as ``*x`` is not valid Python syntax,
-whereas ``x[0]`` is.
+        Cython language uses the normal C syntax for C types, including pointers.  It provides
+        all the standard C types, namely ``char``, ``short``, ``int``, ``long``,
+        ``long long`` as well as their ``unsigned`` versions, e.g. ``unsigned int``.
+        The special ``bint`` type is used for C boolean values (``int`` with 0/non-0
+        values for False/True) and ``Py_ssize_t`` for (signed) sizes of Python
+        containers.
+
+        Pointer types are constructed as in C, by appending a ``*`` to the base type
+        they point to, e.g. ``int**`` for a pointer to a pointer to a C int.
+        Arrays use the normal C array syntax, e.g. ``int[10]``, and the size must be known
+        at compile time for stack allocated arrays. Cython doesn't support variable length arrays from C99.
+        Note that Cython uses array access for pointer dereferencing, as ``*x`` is not valid Python syntax,
+        whereas ``x[0]`` is.
 
 Also, the Python types ``list``, ``dict``, ``tuple``, etc. may be used for
 static typing, as well as any user defined :ref:`extension-types`.
-For example::
+For example
+
+.. tabs::
+
+    .. group-tab:: Pure Python
+
+        .. code-block:: python
+
+            def main():
+                foo: list = []
+
+    .. group-tab:: Cython
 
-    cdef list foo = []
+        .. code-block:: cython
+
+            cdef list foo = []
 
 This requires an *exact* match of the class, it does not allow subclasses.
 This allows Cython to optimize code by accessing internals of the builtin class,
 which is the main reason for declaring builtin types in the first place.
 
 For declared builtin types, Cython uses internally a C variable of type ``PyObject*``.
-The Python types int, long, and float are not available for static
-typing and instead interpreted as C ``int``, ``long``, and ``float``
-respectively, as statically typing variables with these Python
-types has zero advantages.
+
+.. Note:: The Python types ``int``, ``long``, and ``float`` are not available for static
+    typing in Cython language and instead interpreted as C ``int``, ``long``, and ``float``
+    respectively, as statically typing variables with these Python
+    types has zero advantages. On the other hand, annotating in Pure Python with
+    ``int``, ``long``, and ``float`` python types will be interpreted as ``PyObject*`` type.
 
 Cython provides an accelerated and typed equivalent of a Python tuple, the ``ctuple``.
-A ``ctuple`` is assembled from any valid C types. For example::
+A ``ctuple`` is assembled from any valid C types. For example
+
+.. tabs::
+
+    .. group-tab:: Pure Python
+
+        .. code-block:: python
 
-    cdef (double, int) bar
+            def main():
+                bar: (cython.double, cython.int)
```

Comment:
Must be module-level, please remove main

Reply:
See https://github.com/cython/cython/pull/4242#discussion_r669362405